src/main/java/com/shopify/minishopify/controllers/CategoryController.java

```diff
@@ -0,0 +1,58 @@
+package com.shopify.minishopify.controllers;
+
+import com.shopify.minishopify.model.Category;
+import com.shopify.minishopify.model.Shop;
+import com.shopify.minishopify.repository.CategoryRepository;
+import com.shopify.minishopify.repository.ShopRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/category")
+public class CategoryController {
+    @Autowired
+    private CategoryRepository categoryRepository;
+    @Autowired
+    private ShopRepository shopRepository;
+
+    @GetMapping("/list")
+    public List categoryList(@RequestParam(name = "keyword", required = false) String keyword){
+        if(keyword != null){
+            if(!keyword.equals("")) {
+                return categoryRepository.search(keyword);
+            }
+        }
+        return categoryRepository.findAll();
+    }
+
+    @PutMapping("/add/{id}")
+    public Shop addCategoryToShop(@PathVariable int id, @RequestParam(name = "category") String categoryStr) {
```

Comment:
How would that avoid erroneous categories? All it would do is make our JS call our API twice (once to create then after to add)

Reply:
I guess it would be avoided depending on how you were planning on developing the frontend. Do you have an idea right now that you could share?

When I first saw the code, I speculated that the implementation you were going for would be when the user wishes to add a category to a shop, they would type the name of category. If it happened to be a mistyped category, it would still be accepted since it would just create a new category. But thinking about it now that might not necessarily be the case.

Reply:
I'm thinking that there'd be some dropdown with already existing categories/tags, and one of those dropdown elements is a text field that allows you to make a new category/tag

Reply:
The addition of category and tags should be attached as part of the JSON body while creating/updating a shop.

Reply:
Might not make sense to have separate add tag/category operations then.

Reply:
removed this superfluous method. Shop Updates/Create will add tags instead.